### PR TITLE
additions to graphql, fixes to errors

### PR DESCRIPTION
**Pull request by polandll**
More graphql adds, for using collections, udts, tuples, as well as some filters

### CODE REVIEW

File: modules/developers-guide/examples/graphql/DML_data.graphql [hunk 1]
```diff
@@ -1,72 +1,312 @@
...
+    }
+}
+
+# get 3 books using the primary keys with an "in" filter clause of the primary key title
+# *** ADVANCED ***
+query readThreeBooks {
+  book(filter: { title: { in: ["Native Son", "Moby Dick", "Catch-22"] } } ) {
+      values {
+      	title
+ 			  author
```
Comment (dougwettlaufer) [inline modules/developers-guide/examples/graphql/DML_data.graphql:149]: Is this just rendering weird in the diff or is there extra space?

File: modules/developers-guide/examples/graphql/DML_data.graphql [hunk 2]
```diff
@@ -1,72 +1,312 @@
...
+
+# // NEED EXAMPLES FOR USING THE FILTERS: EQ NEQ,GT, GTE,LT, LTE, IN, CONTAINS (FOR MAP?)
+
+# query a badge record that has a MAP (earned) with the partition key and the clustering key
+# *** ADVANCED ***
+query oneGold100Badge {
+  badge(filter: { badge_type: {eq:"Gold"} badge_id: {eq:100}} ) {
+      values {
+      	badge_type
+        badge_id
```
Comment (dougwettlaufer) [inline modules/developers-guide/examples/graphql/DML_data.graphql:193]: Weird diff again?

File: modules/developers-guide/examples/graphql/DML_data.graphql [hunk 3]
```diff
@@ -1,72 +1,312 @@
...
+      	earned {
+          key
+          value
+        }
+     }
+   }
+}
+
+# query a badge record that has a MAP (earned) with the partition key, clustering key, and a MAP key
+# CREATE INDEX badge_idx ON library.badge(KEYS(earned));
```
Comment (dougwettlaufer) [inline modules/developers-guide/examples/graphql/DML_data.graphql:218]: Is the need for an index explained somewhere?

Comment (dougwettlaufer) [reply modules/developers-guide/examples/graphql/DML_data.graphql:218]: Also I'm not familiar, will these comments show up?

File: modules/developers-guide/examples/graphql/DML_data.graphql [hunk 4]
```diff
@@ -1,72 +1,312 @@
...
+          value
+        }
+     }
+   }
+}
+
+# insert a book to use in demonstrating deletebooks
+# *** USED IN QUICKSTART ***
+mutation insertPap {
+  pap: insertbook(value: {title:"Pride and Prejudice", author:"Jane Austen"}) {
```
Comment (dougwettlaufer) [inline modules/developers-guide/examples/graphql/DML_data.graphql:287]: For consistency should this be 'PaP?

**Review by dougwettlaufer — APPROVED** [review]
Few minor comments but looks good

File: modules/developers-guide/examples/graphql/DML_data.graphql (hunk 3 above)
Comment (polandll) [reply modules/developers-guide/examples/graphql/DML_data.graphql:218]: I do explain the indexes in the text. And this file is basically my import/export file for graphql - it isn't in the docs.

Comment (dougwettlaufer) [reply modules/developers-guide/examples/graphql/DML_data.graphql:218]: Got it

**Pull request merged**